types.go

```diff
@@ -236,6 +206,14 @@ type Settings struct {
 	Recorder Recorder
 }
 
+// TODO Comment
```

Comment:
I think we should decide on the full Problem struct now (with constraints), as that way we can hope we don't need more incompatible changes. The two outstanding issues in my mind are how to communicate the number of nonlinear constraints and if Hess et. al. should take matrix interface types instead of concrete types. Thoughts @vladimir-ch ?